### ISSUE TITLE
Failed to compile on android studio

### ISSUE THREAD

**Issue by novalabstudio** [taivo/parse-push-plugin#30]
When i compile with andoird studio, exit with error:

`UNEXPECTED TOP-LEVEL EXCEPTION:
com.android.dex.DexException: Multiple dex files define Landroid/support/annotation/AnimRes;
	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
	at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:502)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:334)
	at com.android.dx.command.dexer.Main.run(Main.java:277)
	at com.android.dx.command.dexer.Main.main(Main.java:245)
	at com.android.dx.command.Main.main(Main.java:106)
`

someone can help me?

**Comment by dyegonery** [comment]
I'm having this issue too.

**Comment by taivo** [comment]
Sounds like you have a duplicate android-support-v4.jar in your project (or android-support-vX.jar where X is an integer). I had the same issue when using this plugin with phonegap-facebook-plugin. Figure out which of your -v4.jar is the latest and keep that one.

If that doesn't work, there is an alternate (but similar) explanation here https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/INSTALLATION.md#multidex

**Comment by FrnandMG** [comment]
@taivo I had the same issue I'm using plugin phonegap-facebook-plugin too. so where exactly is the file that need to delete? I found one in "/PROJECT-FOLDER/platforms/android/libs/android-support-v4.jar"

**Comment by taivo** [comment]
For my case, after installing both the facebook plugin and parse-push-plugin, I had android-support-v4 as well as bolt-android in `platforms/android/libs` and the libs directory under `platforms/android/phonegap-facebook-plugin`. I moved both jar files from `platforms/android/libs` to the second directory and that seemed to work. I moved instead of deleting those files because they are more recent.

In short, you want the latest version of those 2 jar files in facebook plugin's lib dir and delete the copies installed by parse-push-plugin. That worked for me.

**Comment by FrnandMG** [comment]
I found 2 android-support-v4.jar 

one in: `platforms/android/libs`1,3 MB

two in: `platforms/android/phonegap-facebook-plugin/project/FacebookLib/libs/` 758 KB

If deleted the first, parse-push-plugin had an error.

If deleted the second, phonegap-facebook-plugin had an error.

FIX THE ERROR: copying and replacing the number two by the number one.

I had to do the same procedure with `bolts-android.jar`

@taivo thanks for reply.

**Comment by FrnandMG** [comment]
By the way I think my solution works because (I am not java developer expert) `phonegap-facebook-plugin` is compiled first, but it has an old `.jar` for `parse-push-plugin`.

**Comment by campers** [comment]
I have this cordova hook in /hooks/after_prepare to automate deleting it
`#!/usr/bin/env node

/**
 * The facebookconnect and ParsePush plugins both contain a bolts-android-xx.jar
 * Attempting to build for android with duplicate copies will cause a build failure.
 * The facebook one is required for the project to build, so we will delete the one
 * provided by the parse push plugin which is copied to android/libs/.
 */
var exec = require('child_process').exec;
var path = require('path');
var sys = require('sys');
var fs = require('fs');

var rootDir = process.argv[2];
var androidPlatform = rootDir + '/platforms/android/';

// check for the facebook plugin dir as the bolts jar is under a dynamically generated folder containing the app name
var facebookPlugin = androidPlatform + 'com.phonegap.plugins.facebookconnect/'
// The parse plugin copies it bolts-android jar to the android/libs folder
var parsePushBoltsJar = androidPlatform + 'libs/bolts-android-1.1.4.jar'

if( fs.existsSync(facebookPlugin) && fs.existsSync(parsePushBoltsJar)) {
    console.log("---------------------------------------------------------")
    console.log('Detected facebook plugin and parse push bolts-android jar.')
    console.log('Deleting ' + parsePushBoltsJar);
    fs.unlinkSync(parsePushBoltsJar);
    console.log("---------------------------------------------------------")
}`

**Comment by FrnandMG** [comment]
Great, I'm gonna try it.

**Comment by Missile71** [comment]
Sorry to reopen this issue, but I'm using facebook plugin and parse push (I develop with visual studio 2015).
Without facebook all good, but with facebook it seems to use old bolts and support library (from facebook plugin), and getting error not for duplicated library and I didn't find jar from parse push to replace jar from facebook

**Comment by Missile71** [comment]
Thank you @taivo, I found this https://github.com/jeduan/cordova-plugin-facebook4 that use framework instead jar
